### DISCUSSION TITLE
Loading the ISO image to the RAM and perform OS installing

### DISCUSSION THREAD

**Discussion by Naranthiran** [ipxe/ipxe#1302] [General]
Hi,

My requirement is install the OS from the memory.

iPxe config file: 

#!ipxe
initrd http://192.68.32.7/rhellive.iso
kernel http://192.68.32.7/memdisk iso raw

When I execute the ipxe file I am getting the below error.
Could not boot: Exec format error (http://ipxe.org/2e008081)

I also tried to boot with the sanboot command and got an error operation not support.
sanboot http://192.68.32.7/rhellive.iso 

can any one help me to fix this issue, or is there any workaround for my requirement.

Regards
Naranthiran Duraisamy

**Comment by NiKiZe** [comment]
Hi, did you follow and read the URL in the message you are seeing? http://ipxe.org/2e008081

memdisk is a pcbios only binary, my guess is that you are running EFI.

But the real issue, you can't do this, even if you could boot to the ISO, (which is what you do with `sanboot`) once the OS kernel kicks in, everything before this is thrown out.

One way this is handled is thru iSCSI, which `sanboot` works with, where information for how to reach the "disk" is placed in a speciall place in memory. Once the kernel has started the NIC again, and set up IP, it can reconnect iSCSI, but this also requires OS support.

The better way for Linux is to load initrd etc before the kernel boots, but that is diffrent from a ISO boot.

**Comment by Naranthiran** [comment]
Hi @NiKiZe 

Thanks for your quick response. But in my case I do not have a SAN storage, is it still possible to load the ISO with the HTTP based server?

 sanboot http://boot.ipxe.org/freedos/fdfullcd.iso

Regards,
Naranthiran Duraisamy

**Reply by NiKiZe** [reply]
What I explained was which technology exists to do this, iSCSI is required. This is when you have a kernel taking over all devices...

In the case of DOS, there is no kernel, it does require pcbios (don't work in EFI mode) and also won't throw out the handles that iPXE adds to deal with the disc access.

"How do I boot ISO" is a frequent question (try to search issues and discussions for memdisk) and the answer is (almost) always "you don't", how to deal with it depends on what the ISO contains, for Linux you need kernel and initrds separate, depending on the distro you often also need to give squashfs.
